checker/src/abstract_value.rs

```diff
@@ -377,6 +377,40 @@ impl AbstractValueTrait for Rc<AbstractValue> {
         } else if self.is_top() {
             other
         } else {
+            if let Expression::Or { left: x, right: y } = &self.expression {
+                // (x || y) && x == x
+                // (x || y) && y == y
+                if *x == other || *y == other {
+                    return other;
+                }
+                if let Expression::LogicalNot { operand } = &other.expression {
+                    //((x || y) && (!x)) = y
+                    if *x == *operand {
+                        return y.clone();
+                    }
+                    //((x || y) && (!y)) = x
+                    if *y == *operand {
+                        return x.clone();
+                    }
+                }
+            }
+            match (&self.expression, &other.expression) {
+                // (!x && !y) == !(x || y)
```

Comment:
Are you attempting a CNF (instead of the usual DNF)? If so I'm not sure whether this is actually a simplification. I'm guessing the solver may convert this back to DNF.

Reply:
This is a heuristic that does three things:
1) Reduces the number of operators, so the expression is simpler in a strict sense.
2) Enables other simplifications to kick in.
2) Tends to transform expressions that are the ugly results of decompilation from MIR, back into expressions that look like the ones typically written by programmers.

Reply:
I believe DNF is generally better for (3). For (2), pushing ! to the leaves generally increases potential for simplifications. But totally up to you.

Reply:
My previous experience with simplification also points to DNF as the better option in general. This rule is helpful for debugging right now, but we might get back to it in the future.